ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java

```diff
@@ -307,6 +312,52 @@ void canOptimisticallyImport_shouldBeFalseWhenBlockIsMergeButNotOldEnough() {
     assertThat(forkChoiceUtil.canOptimisticallyImport(store, blockToImport)).isFalse();
   }
 
+  @ParameterizedTest
+  @MethodSource("isShufflingStableConditions")
+  void isShufflingStable(final int slot, final boolean expectedResult) {
+    assertThat(forkChoiceUtil.isShufflingStable(UInt64.valueOf(slot))).isEqualTo(expectedResult);
+  }
+
+  public static Stream<Arguments> isShufflingStableConditions() {
+    // slot , expectedResult
+    ArrayList<Arguments> args = new ArrayList<>();
+    for (int i = 80; i < 100; i++) {
+      // every start of epoch, shuffling is considered not stable
+      args.add(Arguments.of(i, i % 8 != 0));
+    }
+
+    return args.stream();
+  }
```

Comment:
I understand what you did here, but from a readability point of view it isn't as clear as if you had a few entries of different scenarios (similar to what you did on `isFinalizationOkConditions`).

Reply:
its just a lot of copy/paste. i might keep the loop for the false.. will see if i can make it clear still